test/e2e_node/container_lifecycle_test.go

```diff
@@ -398,6 +398,181 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
 		framework.ExpectNoError(results.ExitsBefore(prefixedName(PostStartPrefix, regular1), regular2))
 	})
 
+	ginkgo.When("A pod is running a regular container with restartPolicy=Never", func() {
+
+		regular1 := "regular-1"
+
+		podSpec := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "container-image-update-never",
+			},
+			Spec: v1.PodSpec{
+				RestartPolicy: v1.RestartPolicyNever,
+				Containers: []v1.Container{
+					{
+						Name:  regular1,
+						Image: busyboxImage,
```

Comment:
nit: prefer agnhost to other images:
https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md

@aojea @pohly @dims we could probably do more to highlight this?

Reply:
Do what? Communicating this to devs (like a mail to dev) or something technical (like a linter?)?

Reply:
speaking as a new contributor, if most of the tests use agnhost then I'd continue to use it for most tests without thinking twice. I'm working on a PR for this test file to reuse the `Context` and will also update the images to agnhost.

Reply:
The problem is that we need to inform new and existing contributors about such best practices. That's hard, in particular when existing code and tests don't follow them for historic reasons :cry:

Reply:
> The problem is that we need to inform new and existing contributors about such best practices. That's hard, in particular when existing code and tests don't follow them for historic reasons 😢

Right. I think just an email would be a good start. We have https://github.com/kubernetes/kubernetes/pull/122751 but that doesn't handle additional usage of other currently permitted images from historic reasons.